# Load the Clean Dataset and New Dataset

- **Clean Survey Dataset**: The student survey contains comprehensive information about students studying at BCIT, including demographic details, financial data, lifestyle choices, and time allocation. 
- **2024 Salary By Age Group Dataset**: Contains average salary data based on age groups from Year 2024. (**Fake Data** - file generated by gen_fake_data.ipynb saved in **Scripts Folder**

In [1]:
import pandas as pd

import os
print(os.getcwd())

# Load the clean survey dataset last week: Saved in Clean_Data Folder
df_clean = pd.read_csv("../Clean_Data/clean_survey_2025-02-06_assign4_sosochoi.csv")

# Load the fake new dataset: Saved in Data Folder
df_salary = pd.read_csv("../Data/2024_salarybyagegroup_data.csv")

C:\Users\sodiu\PythonLab\Assign5\Assign5_A_new_Branch\Scripts


# Display the Clean Dataset Before Merging

In [2]:
print("Shape of Clean Data:", df_clean.shape)
df_clean

Shape of Clean Data: (250, 28)


Response Timestamp Program  Gender   Age  Height (cm)  \
0    2019/02/04 11:26:35 am GMT-8    HRMG  Female  22.0       177.80   
1    2019/02/04 11:27:45 am GMT-8    HRMG  Female  24.0       175.00   
2    2019/02/04 11:27:46 am GMT-8    HRMG    Male  24.0       182.88   
3    2019/02/04 11:29:06 am GMT-8    HRMG  Female  34.0         5.00   
4    2019/02/04 11:29:12 am GMT-8    HRMG  Female  24.0       160.00   
..                            ...     ...     ...   ...          ...   
245  2025/01/19 10:24:34 pm GMT-8    BMGT  Female  19.0       182.00   
246  2025/01/20 10:55:29 am GMT-8    BMGT  Female  22.0       157.00   
247  2025/01/20 11:33:05 am GMT-8    OMPT    Male  21.0       169.00   
248  2025/01/20 10:43:04 pm GMT-8    ITMG  Female  23.0       157.00   
249   2025/01/23 9:33:54 am GMT-8    MKTG  Female  18.0       170.00   

     Weight (kg)  Previous Avg Monthly Salary ($)  \
0          110.0                           1400.0   
1           63.0                           2000.0   
2          100.0                           2000.0   
3           61.3                           2500.0   
4           54.0                              0.0   
..           ...                              ...   
245         55.0                              0.0   
246         55.0                              0.0   
247        125.0                              0.0   
248         73.0                              0.0   
249        119.2                              0.0   

     Expected Monthly Salary ($)         Preferred Organization Type  \
0                         4000.0  3. Non-profit or Government Agency   
1                         3750.0                    2. Small Company   
2                         3500.0                    2. Small Company   
3                         3500.0                1. Large Corporation   
4                         3800.0                    2. Small Company   
..                           ...                                 ...   
245                       5500.0                    2. Small Company   
246                       2000.0                1. Large Corporation   
247                          NaN                1. Large Corporation   
248                       5447.0                    2. Small Company   
249                       1040.0                    2. Small Company   

    Live with Parents (Yes/No)  ...  Weekly Alcoholic Drinks Has Tattoo  \
0                           No  ...                      0.0        Yes   
1                          Yes  ...                      0.0        Yes   
2                          Yes  ...                      1.0         No   
3                          Yes  ...                      0.0        Yes   
4                          Yes  ...                      0.0         No   
..                         ...  ...                      ...        ...   
245                        Yes  ...                      0.0         No   
246                        Yes  ...                      1.0        Yes   
247                        Yes  ...                      0.0         No   
248                        Yes  ...                      1.0        Yes   
249                         No  ...                      0.0         No   

     Average Nightly Sleep (hours)  Weekly Social Networking Time (hours)  \
0                              8.0                                   10.0   
1                              8.0                                    7.0   
2                              6.0                                   20.0   
3                              7.0                                   14.0   
4                              6.0                                    6.0   
..                             ...                                    ...   
245                            5.0                                   12.0   
246                            8.0                                    3.0   
247                            7.0                    

# Display the 2024 Salary By Age Group Dataset Before Merging

In [3]:
print("Shape of Fake Data:", df_salary.shape)
df_salary

Shape of Fake Data: (3, 5)


Age Group  Average Salary ($)  Median Salary ($) Salary Range ($)  \
0     20-29              4416.0               4300      3500 - 5000   
1     30-39              5311.0               5200      4500 - 6000   
2     40-49              6786.0               6700      6000 - 8000   

   Employment Rate (%)  
0                   60  
1                   75  
2                   80

# Rename Columns for 2024 Salary By Age Group Dataset 

In [4]:
# Rename columns
df_salary.rename(columns={
    "Average Salary ($)": "Average Salary ($) by Age Group",
    "Median Salary ($)": "Median Salary ($) by Age Group",
    "Salary Range ($)": "Salary Range ($) by Age Group",
    "Employment Rate (%)": "Employment Rate (%) by Age Group"
}, inplace=True)

# Display the DataFrame
print(df_salary)

  Age Group  Average Salary ($) by Age Group  Median Salary ($) by Age Group  \
0     20-29                           4416.0                            4300   
1     30-39                           5311.0                            5200   
2     40-49                           6786.0                            6700   

  Salary Range ($) by Age Group  Employment Rate (%) by Age Group  
0                   3500 - 5000                                60  
1                   4500 - 6000                                75  
2                   6000 - 8000                                80  


# Merge the Datasets

In [5]:
# Merge the datasets
merged_df = pd.merge(df_clean, df_salary, on="Age Group", how="left")
# Display the shape of the merged DataFrame
print("Shape of Merged DataFrame:", merged_df.shape)
merged_df

Shape of Merged DataFrame: (250, 32)


Response Timestamp Program  Gender   Age  Height (cm)  \
0    2019/02/04 11:26:35 am GMT-8    HRMG  Female  22.0       177.80   
1    2019/02/04 11:27:45 am GMT-8    HRMG  Female  24.0       175.00   
2    2019/02/04 11:27:46 am GMT-8    HRMG    Male  24.0       182.88   
3    2019/02/04 11:29:06 am GMT-8    HRMG  Female  34.0         5.00   
4    2019/02/04 11:29:12 am GMT-8    HRMG  Female  24.0       160.00   
..                            ...     ...     ...   ...          ...   
245  2025/01/19 10:24:34 pm GMT-8    BMGT  Female  19.0       182.00   
246  2025/01/20 10:55:29 am GMT-8    BMGT  Female  22.0       157.00   
247  2025/01/20 11:33:05 am GMT-8    OMPT    Male  21.0       169.00   
248  2025/01/20 10:43:04 pm GMT-8    ITMG  Female  23.0       157.00   
249   2025/01/23 9:33:54 am GMT-8    MKTG  Female  18.0       170.00   

     Weight (kg)  Previous Avg Monthly Salary ($)  \
0          110.0                           1400.0   
1           63.0                           2000.0   
2          100.0                           2000.0   
3           61.3                           2500.0   
4           54.0                              0.0   
..           ...                              ...   
245         55.0                              0.0   
246         55.0                              0.0   
247        125.0                              0.0   
248         73.0                              0.0   
249        119.2                              0.0   

     Expected Monthly Salary ($)         Preferred Organization Type  \
0                         4000.0  3. Non-profit or Government Agency   
1                         3750.0                    2. Small Company   
2                         3500.0                    2. Small Company   
3                         3500.0                1. Large Corporation   
4                         3800.0                    2. Small Company   
..                           ...                                 ...   
245                       5500.0                    2. Small Company   
246                       2000.0                1. Large Corporation   
247                          NaN                1. Large Corporation   
248                       5447.0                    2. Small Company   
249                       1040.0                    2. Small Company   

    Live with Parents (Yes/No)  ...  Weekly Study Time (hours)  \
0                           No  ...                       10.0   
1                          Yes  ...                       20.0   
2                          Yes  ...                       20.0   
3                          Yes  ...                       20.0   
4                          Yes  ...                       11.0   
..                         ...  ...                        ...   
245                        Yes  ...                       30.0   
246                        Yes  ...                        4.0   
247                        Yes  ...                        NaN   
248                        Yes  ...                       21.0   
249                         No  ...                       35.0   

    Weekly Work Hours  Traveled Outside North America  Pay Own Tuition  \
0                 0.0                             Yes               No   
1                 0.0                             Yes              Yes   
2                 8.0                             Yes               No   
3                 0.0                             Yes               No   
4                 0.0                             Yes              Yes   
..                ...                             ...              ...   
245              24.0                             Yes               No   
246              16.0                             Yes              Yes   
247               0.0                             Yes               No   
248               0.0                             Yes              Yes   
249              15.0                    

# Explain the Type of Merge Chosen: Left Join

- **Data Retention**: To retain all records from the clean dataset (which has 250 rows) and include corresponding salary information based on age groups.
- **Handling Missing Data**: To ensures that if some rows in the clean dataset do not fall into any of the defined age groups in the salary dataset, those records will still be included, with NaN filling in the salary-related fields.

# Type of NA Values are Created 

- The salary-related fields for individuals not falling into defined age groups will contain NaN values. 
- This provides clarity in the dataset, indicating that while the demographic data is available, the corresponding salary information is missing.
- From the merged dataset, we could see that the age group "Under 20" has missing salary data.

# Save the Merged Dataset

In [6]:
# Save the merged dataset: saved in **"Results" Folder**

merged_df.to_csv("../Results/merged_student_data.csv", index=False)